.github/workflows/tests.yaml

```diff
@@ -0,0 +1,47 @@
+name: tests
+
+on:
+  push:
+    branches: [main]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.12"]
+
+    steps:
+      - uses: actions/checkout@v4
+        
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
+        
+      - name: Install dependencies
+        run: poetry install --no-interaction --no-root
+        
+      - name: Install project
+        run: poetry install --no-interaction
+        
+      - name: Run pytest
+        run: |
+          # source .venv/bin/activate
+          poetry run pytest tests/
+
+      - name: Run pyest-cov
+        run: |
+          poetry run pytest --cov rdfproxy/ --cov-report xml
+          
+      - name: Upload coverage data to coveralls
```

Comment:
I guess we need an account with this service? Otherwise the workflow will likely fail?

Reply:
There is an [acdh-oeaw](https://coveralls.io/github/acdh-oeaw) account and the action should automate almost everything, the only thing required is the `GITHUB_TOKEN`, see the [action's inputs](https://github.com/coverallsapp/github-action?tab=readme-ov-file#inputs).